cpp/src/neighbors/ivf_flat/ivf_flat_helpers.cu

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuvs/neighbors/ivf_flat_helpers.cuh>
+
+#include "../detail/ann_utils.cuh"
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/core/resources.hpp>
+#include <raft/neighbors/detail/div_utils.hpp>
+#include <variant>
+
+namespace cuvs::neighbors::ivf_flat {
+namespace codepacker {
+namespace {
+
+template <typename T>
+__device__ void pack_1(const T* flat_code, T* block, uint32_t dim, uint32_t veclen, uint32_t offset)
+{
+  // The data is written in interleaved groups of `index::kGroupSize` vectors
+  using interleaved_group = raft::neighbors::detail::div_utils<kIndexGroupSize>;
```

Comment:
We shouldn't be calling anything in raft::neighbors anymore.

Reply:
Where should I port over `raft::neighbors::detail::div_utils` to?

Reply:
I would move it over to the corresponding place in cuvs.